docs/extension.md

```diff
@@ -0,0 +1,78 @@
+# How to extend this extension
+
+This extension supports only common built-in methods for resolving paths to latte files, collecting variables, components, forms etc.
+However, we know that in a wild there are applications which use their own methods for these purposes. That's why we create this extension extensible.
+
+## Variable collectors
+VariableCollector is a service used to collect variables which can be used in compiled template.
+It uses several sub collectors which basically finds:
+- assigns to [$this->template->foo = 'bar';](../src/LatteContext/Collector/VariableCollector/AssignToTemplateVariableCollector.php)
+- calls [$this->template->setParameters([...]);](../src/LatteContext/Collector/VariableCollector/SetParametersToTemplateVariableCollector.php)
+- and more, see all [variable collectors](../src/LatteContext/Collector/VariableCollector)
+
+You can implement your own variable collector implementing [`VariableCollectorInterface`](../src/LatteContext/Collector/VariableCollector/VariableCollectorInterface.php).
+Then register it as a new service in config file.
+
+## Template path collectors
+TemplatePathCollector is a service which is used to find path to latte templates.
+Now there is only one collector which finds `$template->setTemplate($path)`. So if you use some other way how to tell where the latte template is, feel free to implement [`TemplatePathCollectorInterface`](../src/LatteContext/Collector/TemplatePathCollector/TemplatePathCollectorInterface.php). Don't forget to register this new service in config file.
```

Comment:
It is setFile. Also there shoukd be explanation how it is different from temate renders i think

Reply:
nice catch! thanks